### PR TITLE
Update windows bindings and use windows-targets

### PR DESCRIPTION
**Pull request by ChrisDenton**
This synchronizes the API definitions with std. Trouble is, std uses its own definition of the `windows_targets::link!` macro so I've used a path hack to include it in this crate when used as a dependency of std. Otherwise the `windows-targets` crate is used.

Not sure if this will work but let's see what CI says.

### CODE REVIEW

**Comment by ChrisDenton** [comment]
Oh, path hacks might not work because it looks like there's a difference between fake-std and real-std.

**Comment by ChrisDenton** [comment]
Hmm... maybe I could make `windows_targets` crate in std's sysroot and share it that way.

**Comment by ChrisDenton** [comment]
Ok, with the std changes this should be ready for review now so removing the draft thingy

**Comment by workingjubilee** [comment]
merge conflicts

**Comment by workingjubilee** [comment]
Remember that libstd does this for backtrace-rs:
```rust
    println!("cargo:rustc-cfg=backtrace_in_libstd");
```

File: Cargo.toml [hunk 1]
```diff
@@ -38,6 +38,7 @@ serde = { version = "1.0", optional = true, features = ['derive'] }
 cpp_demangle = { default-features = false, version = "0.4.0", optional = true, features = [
   "alloc",
 ] }
+windows-targets = "0.52.6"
```
Comment (workingjubilee) [inline Cargo.toml:41]: shouldn't this be gated behind `cfg(windows)`?

Comment (ChrisDenton) [reply Cargo.toml:42]: Done!

**Review by workingjubilee — APPROVED** [review]
Thanks! Syncing with std is generally desirable so,

**Pull request merged**